cmd/vcfFilter/vcfFilter.go

```diff
@@ -16,11 +17,34 @@ import (
 
 func vcfFilter(infile string, outfile string, c criteria, groupFile string, parseFormat bool, parseInfo bool, setSeed int64) (total, removed int) {
 	rand.Seed(setSeed)
-	records, header := vcf.GoReadToChan(infile)
+	var records <-chan vcf.Vcf
+	var header vcf.Header
+	var err error
+	var mapContains bool
+	var sitesSeen map[string]map[int]uint8 = make(map[string]map[int]uint8, 0) //uint8 is the number of times this site is seen in the vcf file.
+
+	if c.biAllelicOnly {
+		records, _ = vcf.GoReadToChan(infile)
+		for v := range records {
+			if _, mapContains = sitesSeen[v.Chr]; mapContains { //if an entry in the map for the current chrom exists
+				if _, mapContains = sitesSeen[v.Chr][v.Pos]; mapContains {
```

Comment:
maps are pretty smart when it comes to memory allocation. If you check a map for a key that is not present it will return the zero value for the relevant type so you don't need to worry about checking if it is present then allocating (at least for data types without pointers). So you could replace lines 30-34 with just `sitesSeen[v.Chr][v.Pos]++`